src/Common/Commands.Common/ProfileClient.cs

```diff
@@ -0,0 +1,1158 @@
+                }
+                else
+                {
+                    var storedTenants = account.GetPropertyAsArray(AzureAccount.Property.Tenants);
+                    if (account.Type == AzureAccount.AccountType.User && storedTenants.Count() == 1)
+                    {
+                        TracingAdapter.Information(Resources.AuthenticatingForSingleTenant, account.Id, storedTenants[0]);
+                        AzureSession.AuthenticationFactory.Authenticate(account, environment, storedTenants[0], password,
+                            promptBehavior);
+                    }
+                }
+            }
+            catch (AadAuthenticationException aadEx)
+            {
+                WriteOrThrowAadExceptionMessage(aadEx);
+                return new AzureSubscription[0];
+            }
+
+            try
+            {
+                tenants = tenants ?? account.GetPropertyAsArray(AzureAccount.Property.Tenants);
+                List<AzureSubscription> rdfeSubscriptions = ListServiceManagementSubscriptions(account, environment, 
+                    password, ShowDialog.Never, tenants).ToList();
+
+                // Set user ID
+                foreach (var subscription in rdfeSubscriptions)
+                {
+                    account.SetOrAppendProperty(AzureAccount.Property.Subscriptions, subscription.Id.ToString());
+                }
+
+                if (rdfeSubscriptions.Any())
+                {
+                    return rdfeSubscriptions;
+                }
+                else
+                {
+                    return new AzureSubscription[0];
+                }
+            }
+            catch (AadAuthenticationException aadEx)
+            {
+                WriteOrThrowAadExceptionMessage(aadEx);
+                return new AzureSubscription[0];
+            }
+        }
+
+        private string[] LoadAccountTenants(AzureAccount account, AzureEnvironment environment, SecureString password, ShowDialog promptBehavior)
+        {
+            var commonTenantToken = AzureSession.AuthenticationFactory.Authenticate(account, environment,
+                AuthenticationFactory.CommonAdTenant, password, promptBehavior);
+
+            using (SubscriptionClient SubscriptionClient = AzureSession.ClientFactory
+                        .CreateCustomClient<SubscriptionClient>(
+                            new TokenCloudCredentials(commonTenantToken.AccessToken),
+                            environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ServiceManagement)))
+            {
+                var subscriptionListResult = SubscriptionClient.Subscriptions.List();
+                return subscriptionListResult.Subscriptions.Select(s => s.ActiveDirectoryTenantId).Distinct().ToArray();
+            }
+        }
+
+        private AzureSubscription MergeSubscriptionProperties(AzureSubscription subscription1, AzureSubscription subscription2)
```

Comment:
We still need this as we merge between local subscription properties and remote subscription properties